client/components/date-range/test/index.js

```diff
@@ -129,77 +138,57 @@ describe( 'DateRange', () => {
 
-			const expected = {
-				startDateText: 'MM/DD/YYYY',
-				endDateText: 'MM/DD/YYYY',
-			};
-
-			expect( dateRangeTrigger.props() ).toEqual( expect.objectContaining( expected ) );
+			expect( rangeText ).toEqual( 'MM/DD/YYYY - MM/DD/YYYY' );
 		} );
 
-		test( 'should update trigger props to match currently selected dates', () => {
-			const wrapper = shallow( <DateRange translate={ translate } moment={ moment } /> );
+		test( 'should update trigger props to match currently selected dates', async () => {
+			render( <DateRange translate={ translate } moment={ moment } /> );
 
-			const expectedStartDate = '2018-04-01';
-			const expectedEndDate = '2018-04-29';
+			await userEvent.click( screen.getByLabelText( 'Select date range' ) );
 
-			const newStartDate = moment( expectedStartDate );
-			const newEndDate = moment( expectedEndDate );
+			const fromInputEl = screen.getByLabelText( 'From' );
+			const toInputEl = screen.getByLabelText( 'To' );
+			const applyBtnEl = screen.getByText( 'Apply' );
 
-			// Select dates using API
-			// note: not usually recommended to access component API directly
-			// but it's tricky to do this via DOM on a datepicker...
-			wrapper.instance().onSelectDate( newStartDate );
-			wrapper.instance().onSelectDate( newEndDate );
+			await userEvent.type( fromInputEl, '04/01/2018' );
+			await userEvent.type( toInputEl, '04/29/2018' );
 
-			// Force re-render
-			wrapper.update();
+			await userEvent.click( applyBtnEl );
 
-			const dateRangeTrigger = wrapper.find( DateRangeTrigger );
+			const rangeText = screen
+				.getByLabelText( 'Select date range' )
+				.querySelector( 'span' ).innerHTML;
 
-			const expected = {
-				startDateText: dateToLocaleString( newStartDate ),
-				endDateText: dateToLocaleString( newEndDate ),
-			};
-
-			expect( dateRangeTrigger.props() ).toEqual( expect.objectContaining( expected ) );
+			expect( rangeText ).toEqual( '04/01/2018 - 04/29/2018' );
 		} );
 
-		test( 'should toggle popover on trigger click', () => {
-			const wrapper = shallow( <DateRange translate={ translate } moment={ moment } /> );
-
-			const trigger = wrapper.find( DateRangeTrigger );
-
-			let popover;
+		test( 'should toggle popover on trigger click', async () => {
+			render( <DateRange translate={ translate } moment={ moment } /> );
 
 			// Open
-			trigger.props().onTriggerClick();
-
-			wrapper.update();
-
-			popover = wrapper.find( Popover );
+			await userEvent.click( screen.getByLabelText( 'Select date range' ) );
+			const popoverEl = screen.getByRole( 'tooltip' );
 
-			expect( popover.props().isVisible ).toBe( true );
+			expect( popoverEl ).toBeVisible();
 
 			// Close
-			trigger.props().onTriggerClick();
+			const applyBtnEl = screen.getByText( 'Apply' );
+			await userEvent.click( applyBtnEl );
 
-			wrapper.update();
-
-			popover = wrapper.find( Popover );
-
-			expect( popover.props().isVisible ).toBe( false );
+			expect( popoverEl ).not.toBeVisible();
```

Comment:
When testing that an element is there it could be more precise to test with `.not.toBeInTheDocument()`, because we could have the element rendered but have it hidden with `opacity: 0` or `display: none`. WDYT?

Reply:
Yep, that's a very good point! I'll change!

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/63879/commits/b50bb71c4c0bb51094e7b6e4ef076ba3ac5489d1.